solr/prometheus-exporter/conf/solr-alert-rules.yaml

```
@@ -0,0 +1,243 @@
+      annotations:
+        description: Available disk on Solr pod {{ $labels.base_url }} is low; only {{ $value }}% of total disk is available
+        runbook_url: link_to_runbook_for_this_problem
+    - alert: SolrHighHeapUsage
+      expr: sum by (namespace,base_url) (solr_metrics_jvm_memory_heap_bytes{item="used"}) / sum by (namespace,base_url) (solr_metrics_jvm_memory_heap_bytes{item="max"}) > 0.9
+      for: 5m
+      labels:
+        severity: major
+        impact: node health
+      annotations:
+        description: Solr pod {{ $labels.base_url }} high heap usage at {{ $value }}%
+        runbook_url: link_to_runbook_for_this_problem
+    - alert: SolrHighCPU
+      expr: max by (namespace,base_url) (solr_metrics_jvm_os_cpu_load{item="systemCpuLoad"}) > 20
+      for: 10m
+      labels:
+        severity: major
+        impact: node health
+      annotations:
+        description: Solr pod {{ $labels.base_url }} high system CPU usage {{ $value }}%
+        runbook_url: link_to_runbook_for_this_problem
+    - alert: SolrHighCPULoadAvg
+      expr: max by (namespace,base_url) (solr_metrics_jvm_os_load_average) > 20
+      for: 10m
+      labels:
+        severity: major
+        impact: node health
+      annotations:
+        description: Solr pod {{ $labels.base_url }} high system CPU load {{ $value }}%
+        runbook_url: link_to_runbook_for_this_problem
+    - alert: SolrBlockedOrDeadlockedThreads
+      expr: max by (namespace,base_url) (solr_metrics_jvm_threads{item=~"blocked|deadlock"}) > 0
+      for: 2m
+      labels:
+        severity: critical
+        impact: node health
+      annotations:
+        description: Solr pod {{ $labels.base_url }} has blocked / deadlocked threads
+        runbook_url: link_to_runbook_for_this_problem
+    - alert: SolrUnbalancedLoad
+      # compares the number of requests per base_url to the expected balanced load 1/N (+10% for flexibility)
+      expr: 100 * (sum by(namespace,base_url) (solr_metrics_jetty_requests_total) / ignoring(namespace,base_url) group_left() sum(solr_metrics_jetty_requests_total) > ignoring(namespace,base_url) group_left() max(1 / solr_collections_live_nodes) + 0.1)
+      for: 15m
+      labels:
+        severity: major
+        impact: node performance
+      annotations:
+        description: Unbalanced load ({{ $value }}% of total requests) sent to Solr pod {{ $labels.base_url }}
+        runbook_url: link_to_runbook_for_this_problem
+
+  - name: SolrClusterHealth
+    rules:
+    - alert: SolrZkEnsembleBelowQuorum
+      expr: count(solr_zookeeper_nodestatus == 1) / sum(solr_zookeeper_ensemble_size) <= 0.5
+      for: 2m
+      labels:
+        severity: critical
+        impact: cluster health
+      annotations:
+        description: Healthy Zookeeper node count is below Quorum {{ $value }}
+        runbook_url: link_to_runbook_for_this_problem
+    - alert: SolrZkEnsembleStatus
+      expr: count by (status) (solr_zookeeper_status{status!="green"}) > 0
```

```suggestion
      expr: count by (namespace,status,cluster_id,zk_host) (solr_zookeeper_status{status!="green"}) > 0
```

Need more context for this rule